open_rarity/resolver/opensea_api_helpers.py

```diff
@@ -86,9 +92,11 @@ def fetch_opensea_assets_data(slug: str, token_ids: list[int], limit=30):
             f"[Opensea] Failed to resolve assets for {slug}."
             f"Received {response.status_code}: {response.reason}. {response.json()}"
         )
-        raise Exception(f"[Opensea] Failed to resolve assets with slug {slug}")
+        response.raise_for_status()
 
-    return response.json()["assets"]
+    assets = response.json()["assets"]
+    assets.sort(key=(lambda a: int(a["token_id"])))
+    return assets
```

Comment:
More of an opinion but maybe replace these two lines with
```python
return sorted(assets, key=(lambda a: int(a["token_id"])))
```

Reply:
You also do the following since you aren't checking anything in the response.

```python
return sorted(response.json()["assets"], key=(lambda a: int(a["token_id"])))
```